### ISSUE TITLE
Upgrade plans for glTF 2.0

### ISSUE THREAD

**Issue by donmccurdy** [aframevr/aframe#2693]
I think GLTF2Loader will be ready for A-Frame v0.7.0, but I'm not sure COLLADA2GLTF and the Blender exporter will be updated to glTF2.0 quickly enough. We may not want to bump `gltf-model` until the workflows are stabilized. There is also a tool for upgrading glTF1.0 assets to glTF2.0, so if that's reliable then maybe we don't have to wait on other stuff. We could also have both `gltf-model` and `gltf2-model`, but since three.js is planning to drop GLTFLoader eventually, I think that makes less sense.

For now, I have a `gltf-model-next` component in aframe-extras.

**Comment by takahirox** [comment]
glTF 2.0 has been released!
https://www.khronos.org/news/press/khronos-releases-gltf-2.0-specification

**Comment by donmccurdy** [comment]
Yay!

I think my vote would be:

- Plan to upgrade to three.js r85 for A-Frame 0.7.0.
- Update `gltf-model` from GLTFLoader to GLTF2Loader then, without renaming the component.

Waiting for 0.7.0 also buys some time for the tooling, since COLLADA2GLTF updates haven't been merged yet.

**Comment by sirkitree** [comment]
Just a note that I recently used the gltf 2.0 exporter for blender and found using the core `gltf-model` component doesn't work currently, subsequently found this thread.

Has there been any work on getting the 2.0 spec working with `gltf-model` yet?

**Comment by mokargas** [comment]
@sirkitree You can use `gltf-model-next` from @donmccurdy A-Frame Extras package https://github.com/donmccurdy/aframe-extras/blob/master/src/loaders/gltf-model-next.js

**Comment by takahirox** [comment]
And one more thing.
Make sure to use the latest Blender glTF exporter version.
Recently it's fixed a skinning bug.

**Comment by donmccurdy** [comment]
Quick recap of my thoughts on timing here:

1. THREE.GLTF2Loader is stable already, available in three.js r85 and r86.
2. If we swap GLTF2Loader for GLTFLoader under the hood in `gltf-model`, that will be a breaking change. glTF1.0 models will no longer be compatible. Is that OK? Shall I just create a `gltf-model-legacy` component in aframe-extras, similar to the `gltf-model-next` component I'm maintaining now?
3. glTF blender exporter supports glTF 2.0, COLLADA2GLTF does not yet. I'm hopeful that COLLADA2GLTF can be ready by the time a-frame 0.7.0 would be released.
4. Does upgrading to r85 or r86 mean we need to migrate from VREffect to ArrayCamera (or something)?

tl;dr my vote is to add glTF2.0 in A-Frame 0.7.0, IFF there is not some blocking issue related to VREffect deprecation in new versions of three.js.

**Comment by ngokevin** [comment]
I forget which version of three.js removes the VREffect, but just doing a straight upgrade to GLTF2 for 0.7.0 and having a legacy one somewhere sounds good to me.

**Comment by fernandojsg** [comment]
re ArrayCamera was introduced in r85 but is still not ready for production as, for example, we don't have frustum culling working properly, hopefully that would be solved on r87.
+1 for upgrading to gltf2 and leave v1 as legacy.

**Comment by donmccurdy** [comment]
So if we upgrade to r86 now, we can continue using VREffect as usual?

**Comment by dmarcos** [comment]
@donmccurdy I think so.

**Comment by dmarcos** [comment]
If `renderer.vrEnabled === false` everything should behave as it does now.

**Comment by fernandojsg** [comment]
I think so too, but we need to do some testing because the `WebVRManager` was introduced on the renderer itself and many things rely on that, like setting the viewport, calling rAF, so I don't know if it could have side effects if using vreffect

**Comment by feiss** [comment]
I feel Blender's glTF2 exporter more stable than the old one, I think people will have less troubles with glTF2 and will not need COLLADA at all. Also, the development of the exporter is very active and I think @UX3D-nopper will respond promptly.

**Comment by UX3D-nopper** [comment]
Khronos will sooner or later get in touch with the Blender community. Goal is to integrate the glTF 2.0 exporter into the offical Blender repository.
Another goal is, to replace the current material nodes with an internal Blender node, that the mentioned confusion is elminated. Optimal would be a "glTF 2.0 node" inside Blender, which is using Eevee.
